churaverse-plugins-server/src/churarenPlugin/churarenBossAttackPlugin/churarenBossAttackPlugin.ts

```diff
@@ -0,0 +1,109 @@
+    this.bus.subscribeEvent('registerMessage', socketController.registerMessage.bind(socketController))
+    this.bus.subscribeEvent(
+      'registerMessageListener',
+      socketController.setupMessageListenerRegister.bind(socketController)
+    )
+    this.bus.subscribeEvent('registerOnOverlap', this.registerOnOverlap.bind(this))
+  }
+
+  protected subscribeGameEvent(): void {
+    super.subscribeGameEvent()
+    this.bus.subscribeEvent('update', this.update)
+    this.bus.subscribeEvent('bossAttackRequest', this.generateBossAttack)
+    this.bus.subscribeEvent('entitySpawn', this.onSpawnBossAttack.bind(this))
+  }
+
+  protected unsubscribeGameEvent(): void {
+    super.unsubscribeGameEvent()
+    this.bus.unsubscribeEvent('update', this.update)
+    this.bus.unsubscribeEvent('bossAttackRequest', this.generateBossAttack)
+    this.bus.unsubscribeEvent('entitySpawn', this.onSpawnBossAttack.bind(this))
+  }
+
+  protected handleGameStart(): void {}
+
+  protected handleGameTermination(): void {
+    this.bossAttackPluginStore.bossAttacks.clear()
+  }
+
+  private readonly update = (ev: UpdateEvent): void => {
+    moveBossAttacks(ev.dt, this.bossAttackPluginStore.bossAttacks, this.mapPluginStore.mapManager.currentMap)
+    removeDieBossAttack(this.bossAttackPluginStore.bossAttacks, (bossAttackId: string) => {
+      const bossAttackHitMessage = new BossAttackHitMessage({ bossAttackId })
+      const networkPluginStore = this.store.of('networkPlugin')
+      networkPluginStore.messageSender.send(bossAttackHitMessage)
+    })
+  }
+
+  private init(): void {
+    initBossAttackPluginStore(this.store)
+    this.bossAttackPluginStore = this.store.of('bossAttackPlugin')
+    this.mapPluginStore = this.store.of('mapPlugin')
+  }
+
+  private readonly generateBossAttack = (ev: BossAttackRequestEvent): void => {
+    const bossPluginStore = this.store.of('bossPlugin')
+    const boss = bossPluginStore.bosses.get(ev.bossId)
+    const churarenGameInfo = this.store.of('gamePlugin').games.get(this.gameId)
+    const networkPluginStore = this.store.of('networkPlugin')
+    if (churarenGameInfo === undefined || boss === undefined) return
+
+    // ボスの攻撃をフロントに送信
+    sendSpawnedBossAttack(networkPluginStore.messageSender, this.bus, boss.position, boss.bossId)
+  }
+
+  private readonly registerOnOverlap = (ev: RegisterOnOverlapEvent): void => {
+    ev.collisionDetector.register(
+      this.bossAttackPluginStore.bossAttacks,
+      this.store.of('playerPlugin').players,
+      this.bossAttackHitPlayer.bind(this)
+    )
+  }
+
+  private readonly onSpawnBossAttack = (ev: EntitySpawnEvent): void => {
+    if (!(ev.entity instanceof BossAttack)) return
+    const bossAttack = ev.entity
+    this.bossAttackPluginStore.bossAttacks.set(bossAttack.bossAttackId, bossAttack)
+    bossAttack.attack(this.mapPluginStore.mapManager.currentMap)
+  }
+
+  private bossAttackHitPlayer(bossAttack: BossAttack, player: Player): void {
+    // プレイヤーと衝突したボスの攻撃は消える
```

Comment:
死亡しているプレイヤーにダメージイベントが行かないようにアーリーリターンした方が良さそうです

```suggestion
    if (player.isDead) return
    // プレイヤーと衝突したボスの攻撃は消える
```